ui/dashboards/src/components/TimeRangeControls/TimeRangeControls.tsx

```diff
@@ -41,5 +43,26 @@ export function TimeRangeControls() {
       });
     }
   }
-  return <DateTimeRangePicker timeOptions={TIME_OPTIONS} value={timeRange} onChange={setTimeRange} />;
+
+  const handleRefresh = () => {
```

Comment:
Hmm, do you think we should add a method to the `useTimeRange` hook call `refresh` instead? That way, any component can easily call refresh without having this be specific to our TimeRangeControls

```tsx
const { timeRange, refresh, setTimeRange } = useTimeRange()

return <Button onClick={refresh}>Refresh</Button>
```

Calling refresh would increment a counter known as the `refreshKey` to signal signal components to refetch their queries.

I implemented this concept in the [framework prototype](https://codesandbox.io/s/dashboard-framework-concept-v2-g0qtc7?file=/src/framework/core/time.tsx:527-565) to get a general idea

Reply:
ooh thanks i like this idea, and it makes sense if we also want to refresh for absolute time range, i'll try it!